checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -80,6 +97,84 @@ public ResourceLeakAnnotatedTypeFactory(final BaseTypeChecker checker) {
     this.postInit();
   }
 
+  /**
+   * Collects all the possible owning fields enclosed by the {@link ClassTree} and initializes the
+   * fieldsToFinalizers map.
+   *
+   * @param classTree a classTree
+   * @return a map storing empty-set for possible owning fields
+   */
+  private Map<Element, Set<Element>> CollectFields(ClassTree classTree) {
+    List<? extends Tree> members = classTree.getMembers();
+    Map<Element, Set<Element>> fieldsToFinalizers = new HashMap<>();
+    for (Tree tree : members) {
+      Element elt = TreeUtils.elementFromTree(tree);
+      if (elt != null && isElementPossibleOwningField(elt)) {
+        fieldsToFinalizers.put(elt, new LinkedHashSet<>());
+      }
+    }
+    return fieldsToFinalizers;
+  }
+
+  /**
+   * Is the given element a final field with non-empty @MustCall obligation?
+   *
+   * @param element a element
+   * @return true iff the given element is a final field with non-empty @MustCall obligation
+   */
+  private boolean isElementPossibleOwningField(Element element) {
+    return (element.getKind().isField()
+        && ElementUtils.isFinal(element)
+        && !getMustCallValue(element).isEmpty());
+  }
+
+  @Override
+  public void setRoot(@Nullable CompilationUnitTree root) {
+    super.setRoot(root);
+  }
+
+  /**
+   * Getter method to get the fieldToFinalizers.
+   *
+   * @return the fieldToFinalizers field
+   */
+  public Map<Element, Set<Element>> getFieldToFinalizers() {
+    return fieldToFinalizers;
+  }
+
+  @Override
+  public void preProcessClassTree(ClassTree classTree) {
+    fieldToFinalizers.clear();
+    this.fieldToFinalizers = CollectFields(classTree);
+    super.preProcessClassTree(classTree);
+  }
+
+  @Override
+  public void postProcessClassTree(ClassTree tree) {
+    if (checker.hasOption("infer") && !fieldToFinalizers.isEmpty()) {
```

Comment:
Rather than checking this directly, you can check whether `getWholeProgramInference() == null` (if it is null, WPI is disabled).